src/Core/Exception/Exception.php

```diff
@@ -92,6 +92,8 @@ public function getAttributes(): array
      *   array of "header name" => "header value"
      * @param string|null $value The header value.
      * @return array|null
+     * @deprecated 4.2.0 Use `HttpException::setHeaders()` instead. Response headers
+     *   should be set for HttpException only.
```

Comment:
Should this emit a warning? We could automate replacing this with rector. I'd also like to avoid spending weeks adding warnings in the future.

Reply:
I wasn't sure. I can add a warning that would require users to either use setHeaders(), but if they are not using HttpException for some reason, they need to switch to it. Should we detect this and show a custom warning or just a generic warning explaining both.

I could have this code see if the exception is an HttpException and convert the calls, but then the user might have non-HttpExceptions and it would be confusing.

I need to find out if rector can tell me what the parameters are so we can replace it with the right function.

Reply:
Added deprecation notice when setting only.